internal/transport/http2_client.go

```diff
@@ -244,6 +244,9 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 		<-newClientCtx.Done()       // Block until connectCtx expires or the defer above executes.
 		if connectCtx.Err() != nil {
 			// connectCtx expired before exiting the function.  Hard close the connection.
+			if logger.V(logLevel) {
+				logger.Infof("newClientTransport: closing connection due to connect context expiring.")
```

Comment:
Log `connectCtx.Err()` too?  Seems like we might as well.

```go
if err := connectCtx.Err(); err != nil {
  if logger.V() { logger.Infof("newClientTransport: aborting due to connectCtx: %v", err) }
}
```

Reply:
Switched. I was ok with removing connection closure string here because it'll never receive arbitrary frames since the client transport even doesn't get constructed, so no need for the user to know what caused the connection closure as transport system isn't even under test yet.